usecases/monitoring/prometheus.go

```diff
@@ -141,50 +145,102 @@ type PrometheusMetrics struct {
 	T2VRequestsPerBatch   *prometheus.HistogramVec
 }
 
-func NewServerMetrics(namespace string, reg prometheus.Registerer) *ServerMetrics {
+// NewHTPServerMetrics return the ServerMetrics that can be used in any of the grpc or http servers.
+//
+// Example for HTTP servers.
+//
+//	weaviate_request_duration_seconds_sum{instance="node-1", method="DELETE", route="/v1/schema/{className}", status_code="200"}
+//
+// Example For GRPC servers
+//
+//	weaviate_request_duration_seconds_sum{instance="node-1", method="gRPC", route="/grpc.health.v1.Health/Check", status_code="OK"}
+func NewHTTPServerMetrics(namespace string, reg prometheus.Registerer) *HTTPServerMetrics {
 	r := promauto.With(reg)
+	subsystem := "http"
 
-	return &ServerMetrics{
-		TCPActiveConnections: r.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "tcp_active_connections",
-			Help:      "Current number of accepted TCP connections.",
-		}, []string{"protocol"}),
+	return &HTTPServerMetrics{
 		RequestDuration: r.NewHistogramVec(prometheus.HistogramOpts{
 			Namespace: namespace,
+			Subsystem: subsystem,
 			Name:      "request_duration_seconds",
```

Comment:
In prev company we were using this metrics library which allowed dynamic metric prefixing as well, however, this caused as problems as then you could not search for the full metric name in the code directly.

Like:
1. You are exploring some metrics in the Grafana and then want to investigate where and how a certain metric is emitted.
2. You copy the metric name and do a global search in your code repository.
3. Because the metric name was dynamically constructed by containing multiple prefixes you were not able to find it and had to remove parts of the metric name to finally find it.

What I'm saying is not to use this `Subsystem` feature and simply hard code the full metric name directly, like:
```suggestion
			Name:      "http_request_duration_seconds",
```

Reply:
Similar point with the `Namespace` thing as well. Why do we want to have this `weaviate_` prefix? 

1. Either all metrics coming from weaviate database should have this prefix or none. HTTP metrics are not special in any way. If we want all metrics to have it, we should configure that centrally so people would not forget and introduce inconsistencies. 
2. If the concern in mind is the fear that we will not be able to differentiate which service HTTP metrics are coming from in a microservice setup, we should use a label for that.

Reply:
Having a common metric name allows you to build a reusable dashboard where you can switch between multiple service by simple changing a label which service you are interested in without the need to mess with regexp name queries as: `{__name__=~".*http_server_handled_total"}` and improves metric name predictability/discoverability as well.

Reply:
okay, that makes sense 👍

Reply:
Do you think we need to make the namespace configurable in the function name though?
```go
func NewHTTPServerMetrics(namespace string, reg prometheus.Registerer) *HTTPServerMetrics {
```